apps/dash-manufacture-spc-dashboard/app.py

```diff
@@ -1025,138 +1024,27 @@ def show_current_specs(n_clicks, dd_select, store_data):
-        Output(params[2] + suffix_ooc_g, "value"),
-        Output(params[2] + suffix_indicator, "color"),
-    ],
-    inputs=[Input("interval-component", "n_intervals")],
-    state=[State("value-setter-store", "data")],
-)
-def update_param2_row(interval, stored_data):
-    count, ooc_n, ooc_g_value, indicator = update_count(
-        interval, params[2], stored_data
-    )
-    spark_line_data = update_sparkline(interval, params[2])
-    return count, spark_line_data, ooc_n, ooc_g_value, indicator
-
-
-@app.callback(
-    output=[
-        Output(params[3] + suffix_count, "children"),
-        Output(params[3] + suffix_sparkline_graph, "extendData"),
-        Output(params[3] + suffix_ooc_n, "children"),
-        Output(params[3] + suffix_ooc_g, "value"),
-        Output(params[3] + suffix_indicator, "color"),
-    ],
-    inputs=[Input("interval-component", "n_intervals")],
-    state=[State("value-setter-store", "data")],
-)
-def update_param3_row(interval, stored_data):
-    count, ooc_n, ooc_g_value, indicator = update_count(
-        interval, params[3], stored_data
-    )
-    spark_line_data = update_sparkline(interval, params[3])
-    return count, spark_line_data, ooc_n, ooc_g_value, indicator
-
-
-@app.callback(
-    output=[
-        Output(params[4] + suffix_count, "children"),
-        Output(params[4] + suffix_sparkline_graph, "extendData"),
-        Output(params[4] + suffix_ooc_n, "children"),
-        Output(params[4] + suffix_ooc_g, "value"),
-        Output(params[4] + suffix_indicator, "color"),
-    ],
-    inputs=[Input("interval-component", "n_intervals")],
-    state=[State("value-setter-store", "data")],
-)
-def update_param4_row(interval, stored_data):
-    count, ooc_n, ooc_g_value, indicator = update_count(
-        interval, params[4], stored_data
-    )
-    spark_line_data = update_sparkline(interval, params[4])
-    return count, spark_line_data, ooc_n, ooc_g_value, indicator
-
-
-@app.callback(
-    output=[
-        Output(params[5] + suffix_count, "children"),
-        Output(params[5] + suffix_sparkline_graph, "extendData"),
-        Output(params[5] + suffix_ooc_n, "children"),
-        Output(params[5] + suffix_ooc_g, "value"),
-        Output(params[5] + suffix_indicator, "color"),
-    ],
-    inputs=[Input("interval-component", "n_intervals")],
-    state=[State("value-setter-store", "data")],
-)
-def update_param5_row(interval, stored_data):
-    count, ooc_n, ooc_g_value, indicator = update_count(
-        interval, params[5], stored_data
-    )
-    spark_line_data = update_sparkline(interval, params[5])
-    return count, spark_line_data, ooc_n, ooc_g_value, indicator
+# decorator for list of output
+def create_callback(param):
+    def callback(interval, stored_data):
+        count, ooc_n, ooc_g_value, indicator = update_count(interval, param, stored_data)
+        spark_line_data = update_sparkline(interval, param)
+        return count, spark_line_data, ooc_n, ooc_g_value, indicator
```

Comment:
@ohaanika thanks for suggestions on handling callbacks! This works :100:

Reply:
Amazing! 😍